### PR TITLE
Add mux support

### PR DESCRIPTION
**Pull request by overridesh**
Added mux support

### CODE REVIEW

File: mux/detector.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+import (
+	"net/http"
+
+	botdetector "github.com/devopsfaith/krakend-botdetector"
+	krakend "github.com/devopsfaith/krakend-botdetector/krakend"
+	"github.com/luraproject/lura/config"
+	"github.com/luraproject/lura/logging"
+	"github.com/luraproject/lura/proxy"
+	"github.com/luraproject/lura/router/mux"
+	luramux "github.com/luraproject/lura/router/mux"
```
Comment (kpacha) [inline mux/detector.go:12]: duplicated imports

File: mux/detector.go [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+		return handler(d, next)
+	}
+}
+
+func handler(f botdetector.DetectorFunc, next http.HandlerFunc) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		if f(r) {
+			w.WriteHeader(http.StatusForbidden)
+			w.Write([]byte("bot rejected"))
```
Comment (kpacha) [inline mux/detector.go:45]: the gin implementation doesn't write on the response body. the error passed to the `AbortWithError` function is for the logger, not for the client

File: mux/detector.go [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+		next(w, r)
+	}
+}
+
+type middleware struct {
+	detector botdetector.DetectorFunc
+}
+
+// Register checks the configuration and, if required, registers a bot detector middleware at the gin engine
+func Register(cfg config.ExtraConfig, l logging.Logger) mux.HandlerMiddleware {
```
Comment (kpacha) [inline mux/detector.go:57]: If no engine can be provided to the `Register` function, I think we should change the name of the function (maybe to `NewMiddleware`?)

**Review by kpacha — CHANGES_REQUESTED** [review]

File: mux/detector.go (hunk 1 above)
Comment (overridesh) [reply mux/detector.go:11]: My bad, fixed: 0e90922a2b249c7bea5edcb4f45a4ee46335a22d

File: mux/detector.go (hunk 3 above)
Comment (overridesh) [reply mux/detector.go]: 8523715484064e3aeb822cb5161c24c72d4f233f, I renamed it NewMiddleware, at first I thought so but I didn't want to lose the format.

But it makes a lot more sense to launch it with NewMiddleware.

**Review by kpacha — APPROVED** [review]

**Pull request merged**